crates/runtime/src/lib.rs

```diff
@@ -600,6 +650,11 @@ impl Runtime {
 
         let valid_datasets = Self::get_valid_datasets(app, LogErrors(true));
         let mut futures = vec![];
+
+        if !self.initialize_accelerators(&valid_datasets).await {
```

Comment:
This isn't a friendly user experience to just bail if one of the datasets fails initialization - this function could return the list of valid datasets that didn't have any errors, and we continue initializing those.